### PR TITLE
Automatically add initialization registration of xp

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Auto record initialization of experiment functions to use for copy and set them as descriptors. This removes the need to manually register it, and saves bugs.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by jrapin** [comment]
@teytaud this seems to work fine and simplifies a loooot of code :)

File: docs/benchmarking.rst [hunk 1]
```diff
@@ -66,8 +66,11 @@ In this case, please refer to these [guidelines](adding_an_algorithm.md).
...
 
 - keeps the parametrization of the function, used for instantiating the optimizers.
-- keeping a dictionary of descriptors of your function settings through the :code:`_descriptors` attribute,  which is used to create the columns of the data file produced by the experiments
+- keeping a dictionary of descriptors of your function settings through the :code:`descriptors` attribute,  which is used to create the columns of the data file produced by the experiments.
+  Descriptors are automatically filled up based on int/bool/float/str variables you used to initialize the function, but you can add more used :code:`add_descriptors`
 - let's you override methods allowing custom behaviors such as :code:`evaluation_function` called at evaluation time to possibly avoid noise when possible, and :code:`compute_pseudotime` to mock computation time during benchkmarks.
+- implements a :code:`copy` method for creating a new instance, using the same parameters you provided to intialize the function. The initilization of functions may draw random variables
```
Comment (teytaud) [inline docs/benchmarking.rst:72]:
```suggestion
- implements a :code:`copy` method for creating a new instance, using the same parameters you provided to initialize the function. The initilization of functions may draw random variables
```

File: nevergrad/benchmark/additional/example.py [hunk 2]
```diff
@@ -20,11 +20,11 @@ class CustomFunction(ngfuncs.ExperimentFunction):
 
     def __init__(self, offset):
         super().__init__(self.oracle_call, ng.p.Scalar().set_name(""))
-        self.register_initialization(offset=offset)  # to create equivalent instances through "copy"
         self.offset = offset
-        # add your own function descriptors (from base class, we already get "dimension" etc...)
+        # add your own function descriptors if need be with add_descriptors
+        # (from base class, we already get "dimension" etc...
+        #  + all str/int/bool/float parameters such as offet here)
```
Comment (teytaud) [inline nevergrad/benchmark/additional/example.py:26]:
```suggestion
        #  + all str/int/bool/float parameters such as offset here - for all those types, no use explicitly using add_descriptors)
```

**Comment by teytaud** [comment]
Just excellent, we see a beautiful impact on the code which is shorter and more readable.
I'm not sure of the impact on plottings --- should be ok I guess, but that's an educated guess.

**Review by teytaud — APPROVED** [review]

File: nevergrad/benchmark/additional/example.py [hunk 3]
```diff
@@ -20,11 +20,11 @@ class CustomFunction(ngfuncs.ExperimentFunction):
 
     def __init__(self, offset):
         super().__init__(self.oracle_call, ng.p.Scalar().set_name(""))
-        self.register_initialization(offset=offset)  # to create equivalent instances through "copy"
         self.offset = offset
-        # add your own function descriptors (from base class, we already get "dimension" etc...)
+        # add your own function descriptors if need be with add_descriptors
+        # (from base class, we already get "dimension" etc...
+        #  + all str/int/bool/float parameters such as offset here - for all those types, no use explicitly using add_descriptors)
```
Comment (jrapin) [inline nevergrad/benchmark/additional/example.py:26]:
```suggestion
        #  and all str/int/bool/float parameters such as offset here - for all those types, no need to use add_descriptors)
```

**Comment by jrapin** [comment]
> I'm not sure of the impact on plottings --- should be ok I guess, but that's an educated guess.

For plotting this should be transparent, if not let me know!

**Pull request merged**